src/util/rocksdb/mod.rs

```diff
@@ -335,6 +347,61 @@ pub fn compact_range(
     db.compact_range_cf_opt(handle, &compact_opts, start_key, end_key);
 }
 
+pub fn compact_files_in_range(db: &DB, start_key: &[u8], end_key: &[u8]) -> Result<(), String> {
+    for cf_name in db.cf_names() {
+        compact_files_in_range_cf(db, cf_name, start_key, end_key)?;
+    }
+    Ok(())
+}
+
+pub fn compact_files_in_range_cf(
+    db: &DB,
+    cf_name: &str,
+    start_key: &[u8],
+    end_key: &[u8],
+) -> Result<(), String> {
+    let cf = db.cf_handle(cf_name).unwrap();
+    let cf_opts = db.get_options_cf(cf);
+    let output_level = cf_opts.get_num_levels() - 1;
+    let output_compression = cf_opts
+        .get_compression_per_level()
+        .get(output_level)
+        .cloned()
+        .unwrap_or(DBCompressionType::No);
+    let output_file_size_limit = cf_opts.get_target_file_size_base();
+
+    let mut input_files = Vec::new();
+    let mut all_files_in_output_level = true;
+    let cf_meta = db.get_column_family_meta_data(cf);
+    for (i, level) in cf_meta.get_levels().iter().enumerate() {
+        for f in level.get_files() {
+            if !end_key.is_empty() && f.get_smallestkey() >= end_key {
+                continue;
+            }
+            if !start_key.is_empty() && f.get_largestkey() < start_key {
+                continue;
+            }
+            if i != output_level {
+                all_files_in_output_level = false;
+            }
+            input_files.push(f.get_name());
+        }
+    }
+    if input_files.is_empty() || all_files_in_output_level {
+        return Ok(());
+    }
+
+    let mut opts = CompactionOptions::new();
+    opts.set_compression(output_compression);
+    let max_subcompactions = sys_info::cpu_num().unwrap() / 2;
+    let max_subcompactions = cmp::min(max_subcompactions, 16);
```

Comment:
How about the difference in compaction speed when max_subcompaction is 8 and 16?

Reply:
According to the test, one thread can compact in about 30MB/s.
If `max_subcompactions=8`, it will compact in 240MB/s, which may not fully utilize the disk.
If `max_subcompactions=16`, it will compact in 480MB/s, which is fast enough for most disks.